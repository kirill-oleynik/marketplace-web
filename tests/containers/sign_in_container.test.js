const React = require('react');
const renderer = require('react-test-renderer');
const { SignInContainer } = require('../../app/containers/sign_in_container');

jest.mock('../../app/components/auth/sign_in_form', () => 'Form');

describe('#render', () => {
  test('it renders correctly', () => {
    const tree = renderer.create(
      <SignInContainer
        errors={{}}
        signIn={() => {}}
        t={(translation) => translation}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
