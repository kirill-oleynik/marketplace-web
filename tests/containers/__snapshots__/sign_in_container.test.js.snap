// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<div>
  <div
    className="mb-15"
  >
    <h2
      className="login-form__title"
    >
      title
    </h2>
    <span
      className="d-inline-block font-14 mb-10"
    >
      newToAppreview
      <p
        className="login-form__link ml-5"
        onClick={[Function]}
      >
        createAccount
      </p>
    </span>
    <button
      className="button-login-with-socials button-login-with-socials--linkedin w-100 mb-20"
      type="button"
    >
      <i
        className="icon icon-linkedin button-login-with-socials__icon"
      />
      <span>
        Log in with Linkedin
      </span>
    </button>
    <div
      className="d-flex align-items-center justify-content-between mb-10"
    >
      <div
        className="login-form__divider divider divider--dark"
      />
      <span
        className="font-14 mr-20 ml-20"
      >
        or
      </span>
      <div
        className="login-form__divider divider divider--dark"
      />
    </div>
  </div>
  <Form
    errors={Object {}}
    initialState={
      Object {
        "email": "",
        "password": "",
        "rememberMe": false,
      }
    }
    onSubmit={[Function]}
    rules={
      Object {
        "email": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
        "password": Array [
          Array [
            [Function],
            "common:validations.isNotEmpty",
          ],
        ],
        "rememberMe": Array [
          Array [
            [Function],
            "common:validations.isBoolean",
          ],
        ],
      }
    }
    validateOnChange={true}
    validateSingle={true}
  />
</div>
`;
